sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -175,6 +175,27 @@ case class DataSourceResolution(
     case DataSourceV2Relation(CatalogTableAsV2(catalogTable), _, _) =>
       UnresolvedCatalogRelation(catalogTable)
 
+    case ShowTablesStatement(None, pattern) =>
+      defaultCatalog match {
+        case Some(_) =>
+          throw new AnalysisException(
+            "The default namespace cannot be deduced in v2 catalog yet")
+        case None =>
+          ShowTablesCommand(None, pattern)
+      }
+
+    case plan @ ShowTablesStatement(Some(namespace), pattern) =>
+      val CatalogNamespace(maybeCatalog, ns) = namespace
+      maybeCatalog match {
+        case Some(v2Catalog) =>
+          if (ns.isEmpty) {
```

Comment:
what's the expected behavior of `SHOW TABLES catalogName`? Since `[]` is a valid namespace, I think it's reasonable that users want to list the tables under namespace `[]` in `catalogName`.

Reply:
I agree, it should list tables in the empty namespace.

Reply:
OK. I removed this check.